openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceDiagnosticMessages.java

```diff
@@ -33,7 +33,9 @@ public enum ServiceDiagnosticMessages {
             "Query parameters with no array item type can not be mapped to Ballerina resource" +
                     " query parameters.", DiagnosticSeverity.ERROR),
     OAS_SERVICE_102("OAS_SERVICE_102",
-            "Query parameters with type '%s' can not be mapped to the Ballerina query parameters.",
+            "Type '%s' is not a valid query parameter type in Ballerina. " +
+                    "The supported types are string, int, float, boolean, decimal, array types of the aforementioned " +
+                    "types and map<json>.",
```

Comment:
Are we planning to update this message including records when the open record fix is given in update 6 ?

Reply:
As discussed, this implementation won't be excluding the code generation for open records. Ideally, it should be compatible with the HTTP module version including the [fix for open records](https://github.com/ballerina-platform/ballerina-standard-library/issues/4541).

Reply:
I meant that in this message it says that only these types `string, int, float, boolean, decimal` are supported. When are we going to include `record` in that ?

Reply:
Yeah, this error message aligned with the http module's existing message and I checked with them whether we need to modify it with `record`, and they informed that `map<json>` indicates the `record` behaviour so this message won't be modified.